Add accessibility web stack and tests diagrams (#2009)
w3c/aria: documentation/tests.md

# ARIA Testing Documentation

## Some background

The images and text in this section come from James Craig's presentation on the [Accessibility Testing Investigation for WPT Interop 2023](https://github.com/web-platform-tests/interop-2023-accessibility-testing).

![Diagram of the webstack from the web author, to the browser, to the at, to the user, with extra details about the browser process](images/web_accessibility_stack_axapi.png)

> 1. Web Author creates a website with HTML, CSS, and JS that is accessed in a browser.
> 3. In the case the mainstream interface that is accessed directly by those who do not use assistive technologies, the rendering engine (WebKit, Chromium, Gecko) handles all the input/output to and from the end user: input as mouse clicks, keyboard events, etc. and output as pixels rendered on screen, among other things.

Reviewer: ```suggestion
1. Web Author creates a website with HTML, CSS, and JS that is accessed in a browser.
2. In the case the mainstream interface that is accessed directly by those who do not use assistive technologies, the rendering engine (WebKit, Chromium, Gecko) handles all the input/output to and from the end user: input as mouse clicks, keyboard events, etc. and output as pixels rendered on screen, among other things.
```

> 3. In the case the mainstream interface that is accessed directly by those who do not use assistive technologies, the rendering engine (WebKit, Chromium, Gecko) handles all the input/output to and from the end user: input as mouse clicks, keyboard events, etc. and output as pixels rendered on screen, among other things.

Reviewer: Ah I see. Looks like you combined numbers 1 and 2 from the original comment in the issue:

> 1. Web Author (via a number of means) vends Web API that is accessed in a browser.
> 2. Most Web API ends up as HTML, JavaScript, CSS, etc.

👍

> 4. In the case of accessibility technology use such as a screen reader, the engine has internal accessible code (such as WebCore/accessibility in WebKit) that interprets the rendered page and associated Web API, and then vends an equivalent to the associated platform accessibility API. In the case of this diagram, that is represented by macOS's "AX API."
> 5. In most case, assistive technology (AT) such as a screen reader, will access data from the platform API, rather than by inspecting the Web APIs directly, though there are exceptions to this general rule.

Reviewer: ```suggestion
3. In the case of accessibility technology use such as a screen reader, the engine has internal accessible code (such as WebCore/accessibility in WebKit) that interprets the rendered page and associated Web API, and then vends an equivalent to the associated platform accessibility API. In the case of this diagram, that is represented by macOS's "AX API."
4. In most case, assistive technology (AT) such as a screen reader, will access data from the platform API, rather than by inspecting the Web APIs directly, though there are exceptions to this general rule.
```

Reviewer: follow on from previous

> ![Diagram of the webstack from the web author, to the browser, to the at, to the user, with extra details about the browser process](images/web_accessibility_stack_axapi_tests.png)

Reviewer: ```suggestion
![Diagram of the web stack input/output from the web author, to the browser engine, to the engine's accessibility internals, to the platform accessibility API, then to the assistive technology (such as a screen reader), and finally to the user, with additional details about the browser process](images/web_accessibility_stack_axapi_tests.png)
```
